Sources/Basic/Path.swift

```diff
@@ -327,6 +327,51 @@ private struct PathImpl {
 }
 
 
+extension AbsolutePath {
+    /// Returns a relative path that, when concatenated to `base`, yields the
+    /// callee path itself.  If `base` is not an ancestor of the callee, the
+    /// returned path will begin with one or more `..` path components.
+    ///
+    /// Because both paths are absolute, they always have a common ancestor
+    /// (the root path, if nothing else).  Therefore, any path can be made
+    /// relative to any other path by using a sufficient number of `..` path
+    /// components.
+    ///
+    /// This method is strictly syntactic and does not access the file system
+    /// in any way.  Therefore, it does not take symbolic links into account.
+    public func relative(to base: AbsolutePath) -> RelativePath {
```

Comment:
I agree `joined` seems better for Swift3.

Hate seems a strong word... `+` usually maps to some other named operation (not "plus"), so the name for this should be whatever name you would want to use there. I have no objections to a better word than "joined", but there should be a method in any case even if we wanted the operator.